### DISCUSSION TITLE
プリページングの実装

### DISCUSSION THREAD

**Discussion by keisuke713** [nuta/microkernel-book#20] [General]
[プリページング](https://github.com/nuta/microkernel-book/blob/main/IDEAS.md#%E3%83%97%E3%83%AA%E3%83%9A%E3%83%BC%E3%82%B8%E3%83%B3%E3%82%B0%E3%81%AE%E5%AE%9F%E8%A3%85)の実装できました！
最終的な[ソースコード](https://github.com/keisuke713/microkernel-book/commit/710dabf29541e72cdc59c6bf4ec53cac76a9da03)は既存のページフォルト時のコードとほとんど同じですが、実装の過程でロードの流れ・仮想アドレスやページテーブルについて深く知れてとても良い実装アイディアでした！
たくさん質問に答えていただきありがとうございます。
次は共有メモリの実装をやってみようと思います。

**Comment by nuta** [comment]
共有ありがとうございます！大変綺麗な実装で素晴らしいです！横着せずきちんと`p_flags`を見て、必要最低限のページアクセス権限を与えているところが素敵です 💯
